app/controllers/microservices_engine/v1/data_controller.rb

```diff
@@ -34,18 +34,25 @@ def update
         data = params['content']
         if data.present?
           data.each do |endpoint|
-            existing = Connection.where(object: endpoint['object'])
+            existing = Connection.where(object: endpoint['object']).first
             if existing.present?
               if endpoint['url'].present?
                 # URL exists so we will update as usual
-                existing.update_attributes(name: endpoint['name'], url: endpoint['url'])
+                existing.update_attributes(
+                  name: endpoint.require(:name),
+                  url: endpoint.require(:url)
+                )
               else
                 # URL is blank, thus the endpoint no longer exists
```

Comment:
Either way, I can get it done relatively easy but I just want to know how I should go about it.